bot/cogs/antimalware.py

```diff
@@ -9,29 +9,32 @@
 
 
 class AntiMalware(Cog):
-    """Cog providing anti-malware behavior."""
+    """Delete messages which contain attachments with non-whitelisted file extensions."""
 
     def __init__(self, bot: Bot):
         self.bot = bot
-        self.whitelist = tuple(AntiMalwareConfig.whitelist)
 
     @Cog.listener()
     async def on_message(self, message: Message) -> None:
         """Identify messages with prohibited attachments."""
-        rejected_attachments = [a for a in message.attachments if
-                                not a.filename.lower().endswith(self.whitelist)]
-        detected_pyfile = len([a for a in message.attachments if a.filename.lower().endswith('.py')]) > 0
-
-        if len(rejected_attachments) > 0:
+        rejected_attachments = list()
+        detected_pyfile = list()
+        for attachment in message.attachments:
+            if not attachment.filename.lower().endswith(tuple(AntiMalwareConfig.whitelist)):
```

Comment:
Does it really have to be a tuple or can it just be any sequence? Presumably it is was list before you converted it to a tuple.

Reply:
Oddly enough, it does have to be a tuple or str: https://docs.python.org/3/library/stdtypes.html?highlight=endswith#str.endswith